packages/backend/src/open_payments/receiver/service.ts

```diff
@@ -147,10 +148,27 @@ async function getReceiver(
   deps: ServiceDependencies,
   url: string
 ): Promise<Receiver | undefined> {
-  const incomingPayment = await getIncomingPayment(deps, url)
-  if (incomingPayment) {
-    return new Receiver(incomingPayment)
+  try {
+    const localIncomingPayment = await getLocalIncomingPayment({
+      deps,
+      url
+    })
+    if (localIncomingPayment) {
+      return new Receiver(localIncomingPayment, true)
+    }
+
+    const remoteIncomingPayment = await getRemoteIncomingPayment(deps, url)
+    if (remoteIncomingPayment) {
+      return new Receiver(remoteIncomingPayment, false)
+    }
+  } catch (error) {
+    deps.logger.error(
+      { errorMessage: error instanceof Error && error.message },
+      'Could not get incoming payment'
+    )
   }
+
+  return undefined
```

Comment:
The real change here in `getReceiver` is that we create the receiver with `isLocal`. The other changes are a little refactor in support of this. Just moves the attempt to get the local incoming payment into this main  `getReceiver` function so I dont have to return more stuff from `getIncomingPayment` (now `getRemoteIncomingPayment`).